[
    {
        "content": "<p>I've recently (in the past 24 -48 hours) started getting this Jekyll failure:</p>\n<div class=\"codehilite\"><pre><span></span><code>Jekyll: Source: /Users/rhausam/git-repo/fhir-ips-2/temp/pages                    (02:59.0834)\nJekyll: Generating...                                                            (02:59.0834)\nJekyll has failed. Complete output from running Jekyll:   Liquid Exception: Could not locate the included file &#39;fragment-summary-table.html&#39; in any of [&quot;/Users/rhausam/git-repo/fhir-ips-2/temp/pages/_includes&quot;]. Ensure it exists in one of those directories and, if it is a symlink, does not point outside yo (03:00.0464)\nNote: Check that Jekyll is installed correctly                                   (03:00.0464)\nPublishing Content Failed: Process exited with an error: 1 (Exit value: 1)       (03:00.0465)\n                                                                                 (03:00.0465)\nUse -? to get command line help                                                  (03:00.0466)\n                                                                                 (03:00.0466)\nStack Dump (for debugging):                                                      (03:00.0466)\norg.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)\n    at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)\n    at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)\n    at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:153)\n    at org.hl7.fhir.igtools.publisher.Publisher.runJekyll(Publisher.java:5801)\n    at org.hl7.fhir.igtools.publisher.Publisher.runTool(Publisher.java:5723)\n    at org.hl7.fhir.igtools.publisher.Publisher.generate(Publisher.java:5114)\n    at org.hl7.fhir.igtools.publisher.Publisher.createIg(Publisher.java:908)\n    at org.hl7.fhir.igtools.publisher.Publisher.execute(Publisher.java:752)\n    at org.hl7.fhir.igtools.publisher.Publisher.main(Publisher.java:8664)\n</code></pre></div>\n<p>At first it was only occurring in the CI build - so I missed seeing it initially.  But after deleting my local package cache I'm now getting the same error on my local build.<br>\n<span class=\"user-mention\" data-user-id=\"191316\">@Grahame Grieve</span> <span class=\"user-mention\" data-user-id=\"248736\">@Mark Iantorno</span> <span class=\"user-mention\" data-user-id=\"191832\">@Jose Costa Teixeira</span> <span class=\"user-mention\" data-user-id=\"191320\">@Lloyd McKenzie</span></p>",
        "id": 251763147,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630607340
    },
    {
        "content": "<ol>\n<li>did you get today's ig publisher?</li>\n</ol>",
        "id": 251763324,
        "sender_full_name": "Jose Costa Teixeira",
        "timestamp": 1630607401
    },
    {
        "content": "<ol start=\"2\">\n<li>do you have a repo for me to look into it?</li>\n</ol>",
        "id": 251763396,
        "sender_full_name": "Jose Costa Teixeira",
        "timestamp": 1630607414
    },
    {
        "content": "<p>Yes, I've updated to 1.1.79 and that didn't change anything.<br>\n<a href=\"https://github.com/HL7/fhir-ips/tree/connectathon-2\">https://github.com/HL7/fhir-ips/tree/connectathon-2</a></p>",
        "id": 251763929,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630607602
    },
    {
        "content": "<p>first I am finding this issue with the spreadsheet names - is this known yet? </p>\n<div class=\"codehilite\"><pre><span></span><code>Exception generating resource C:\\fhir-ips-connectathon-2\\input\\vocabulary\\valueset-medication-example-uv-ips::ValueSet/medication-example-uv-ips: The workbook already contains a sheet named &#39;Include from National drug codes&#39; (02:57.0919)\njava.lang.IllegalArgumentException: The workbook already contains a sheet named &#39;Include from National drug codes&#39;\n        at org.apache.poi.xssf.usermodel.XSSFWorkbook.validateSheetName(XSSFWorkbook.java:910)\n        at org.apache.poi.xssf.usermodel.XSSFWorkbook.createSheet(XSSFWorkbook.java:863)\n        at org.hl7.fhir.r5.renderers.spreadsheets.ValueSetSpreadsheetGenerator.genIncludeSystem(ValueSetSpreadsheetGenerator.java:117)\n        at org.hl7.fhir.r5.renderers.spreadsheets.ValueSetSpreadsheetGenerator.genInclude(ValueSetSpreadsheetGenerator.java:87)\n        at org.hl7.fhir.r5.renderers.spreadsheets.ValueSetSpreadsheetGenerator.renderValueSet(ValueSetSpreadsheetGenerator.java:40)\n        at org.hl7.fhir.igtools.publisher.Publisher.generateOutputsValueSet(Publisher.java:7815)\n</code></pre></div>",
        "id": 251767908,
        "sender_full_name": "Jose Costa Teixeira",
        "timestamp": 1630609084
    },
    {
        "content": "<p>Yes, there are a bunch of these errors since Grahame added the terminology spreadsheets to the publisher.  But they didn't stop my local builds, until I deleted the package cache in <code>.fhir</code>.  I've been ignoring those errors for the moment.</p>",
        "id": 251768412,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630609257
    },
    {
        "content": "<p>i'm currently stuck. Jekyll hangs</p>",
        "id": 251768815,
        "sender_full_name": "Jose Costa Teixeira",
        "timestamp": 1630609418
    },
    {
        "content": "<p>Yes - that's exactly the issue.</p>",
        "id": 251768878,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630609441
    },
    {
        "content": "<p>(or the thing that comes after Jenkins</p>\n<div class=\"codehilite\"><pre><span></span><code>Jekyll: Generating...                                                            (03:22.0242)\nSending usage stats failed: Connection timed out: connect\n</code></pre></div>",
        "id": 251768933,
        "sender_full_name": "Jose Costa Teixeira",
        "timestamp": 1630609450
    },
    {
        "content": "<p>Oh - different issue. <span aria-label=\"smile\" class=\"emoji emoji-1f642\" role=\"img\" title=\"smile\">:smile:</span></p>",
        "id": 251768993,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630609472
    },
    {
        "content": "<p>On a different IPS branch, I have another issue.  On the 'connectathon' branch (which uses FSH and has more complicated value sets but less slicing), I now get this issue:</p>\n<div class=\"codehilite\"><pre><span></span><code>onLoad:\nRoot directory: /Users/rhausam/git-repo/fhir-ips                                 (00:12.0967)\nCore Package hl7.fhir.r4.core#4.0.1\nTerminology Cache is at /Users/rhausam/git-repo/fhir-ips/input-cache/txcache. 15 files in cache (00:16.0960)\nConnect to Terminology Server at http://tx.fhir.org                              (00:32.0348)\nLoad Package hl7.fhir.pubpack#0.0.9\nLoad Package hl7.fhir.xver-extensions#0.0.5\nLoad Package hl7.terminology.r4#2.1.0\nInitialization complete                                                          (00:34.0314)\nFetch http://hl7.org/fhir/uv/ips/package-list.json for version check             (00:34.0316)\nLoad Content                                                                     (00:34.0459)\nTranslating CQL source                                                           (00:34.0521)\nTranslating CQL source in folder /Users/rhausam/git-repo/fhir-ips/input/cql      (00:34.0521)\nPublishing Content Failed: No Source directories to scan found                   (00:34.0525)\n                                                                                 (00:34.0526)\nUse -? to get command line help                                                  (00:34.0526)\n                                                                                 (00:34.0526)\nStack Dump (for debugging):                                                      (00:34.0526)\norg.hl7.fhir.exceptions.FHIRException: No Source directories to scan found\n    at org.hl7.fhir.igtools.publisher.SimpleFetcher.scan(SimpleFetcher.java:301)\n    at org.hl7.fhir.igtools.publisher.Publisher.loadResources(Publisher.java:3705)\n    at org.hl7.fhir.igtools.publisher.Publisher.load(Publisher.java:3200)\n    at org.hl7.fhir.igtools.publisher.Publisher.createIg(Publisher.java:892)\n    at org.hl7.fhir.igtools.publisher.Publisher.execute(Publisher.java:752)\n    at org.hl7.fhir.igtools.publisher.Publisher.main(Publisher.java:8664)\n</code></pre></div>\n<p>That issue also first appeared in the CI build.  It was not occurring for me locally <strong>until</strong> I deleted the package cache.</p>",
        "id": 251769388,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630609641
    },
    {
        "content": "<p>still stuck. I'm giving up</p>",
        "id": 251772111,
        "sender_full_name": "Jose Costa Teixeira",
        "timestamp": 1630610753
    },
    {
        "content": "<p>Thanks for trying to look at it.  This is a big showstopper for me now.<br>\n<span class=\"user-mention\" data-user-id=\"191316\">@Grahame Grieve</span></p>",
        "id": 251772521,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630610920
    },
    {
        "content": "<p>I’m getting this error in qi-core now as well, but it builds locally without any issues. Not sure what to try next?</p>",
        "id": 252106625,
        "sender_full_name": "Bryn Rhodes",
        "timestamp": 1630881471
    },
    {
        "content": "<p>Missing ‘fragment-summary-table.html’</p>",
        "id": 252106678,
        "sender_full_name": "Bryn Rhodes",
        "timestamp": 1630881501
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"191359\">@Bryn Rhodes</span>  I fixed this issue for my builds by using the <code>#current</code> template (in ig.ini).  Eventually I (we) should be able to move it back, but that was still failing when I last tried it (I think a little over 24 hours ago).</p>",
        "id": 252108666,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630883847
    },
    {
        "content": "<p>or you can clean out your package cache</p>",
        "id": 252112260,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1630888097
    },
    {
        "content": "<p>this won't be an issue after the next release</p>",
        "id": 252112264,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1630888105
    },
    {
        "content": "<p>I just tried removing <code>#current</code> from the template again and deleting the package cache, and when I re-run the build I get the same <code>Liquid Exception: Could not locate the included file 'fragment-summary-table.html' in any of ...</code> message that I've been getting before - <code>#current</code> still seems to be needed for now.<br>\n<span class=\"user-mention\" data-user-id=\"191316\">@Grahame Grieve</span></p>",
        "id": 252113000,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630888931
    },
    {
        "content": "<p>the released package is the same as current at this time</p>",
        "id": 252113043,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1630889022
    },
    {
        "content": "<p>So it shouldn't be different - but it is.</p>",
        "id": 252113110,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1630889094
    },
    {
        "content": "<p>That did the trick for QI-Core as well, thank you <span class=\"user-mention\" data-user-id=\"191405\">@Rob Hausam</span></p>",
        "id": 252115994,
        "sender_full_name": "Bryn Rhodes",
        "timestamp": 1630892571
    }
]